server/src/main/java/org/elasticsearch/search/retriever/LinearRetrieverComponent.java

```
@@ -0,0 +1,92 @@
+        Float weight = null;
+        ScoreNormalizer normalizer = null;
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            if (parser.currentToken() == XContentParser.Token.FIELD_NAME) {
+                if (RETRIEVER_FIELD.match(parser.currentName(), parser.getDeprecationHandler())) {
+                    parser.nextToken();
+                    ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+                    parser.nextToken();
+                    ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser);
+                    final String retrieverName = parser.currentName();
+                    parser.nextToken();
+                    retrieverBuilder = parser.namedObject(RetrieverBuilder.class, retrieverName, context);
+                    parser.nextToken();
+                } else if (WEIGHT_FIELD.match(parser.currentName(), parser.getDeprecationHandler())) {
+                    parser.nextToken();
+                    weight = parser.floatValue();
+                } else if (NORMALIZER_FIELD.match(parser.currentName(), parser.getDeprecationHandler())) {
+                    parser.nextToken();
+                    if (parser.currentToken() == XContentParser.Token.VALUE_STRING) {
+                        normalizer = ScoreNormalizer.valueOf(parser.text());
+                    } else if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
+                        parser.nextToken();
+                        normalizer = ScoreNormalizer.parse(parser.currentName(), parser);
+                        parser.nextToken();
+                    } else {
+                        throw new ParsingException(parser.getTokenLocation(), "Unsupported token [" + parser.currentToken() + "]");
+                    }
+                }
```

Do we want to throw an exception if field_name did not match retriever, weight or normalizer?

---

Yeap, makes sense! Updated.